risc0/bigint2/src/ec/mod.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[cfg(test)]
+mod tests;
+
+use crate::ffi::sys_bigint2_2;
+
+const DOUBLE_BLOB: &[u8] = include_bytes!("double.blob");
+
+pub fn double(point: &[u32], result: &mut [u32]) {
```

Comment:
Not for this PR, but as followup: This is a specific EC double, and we're going to want more than one as we patch multiple crates. So we'll need to genericize the naming and add the alternates (or genericize this one with more parameters).

The EC Double function at https://github.com/risc0/zirgen/blob/040fcd133094974a9a218c13f717c98be0fbada9/zirgen/circuit/bigint/bigint2c.cpp#L420 that this is based on hard-codes the curve parameters to be those of `secp256k1` (given in affine coordinates and Weierstrass form). Note that the underlying function call https://github.com/risc0/zirgen/blob/040fcd133094974a9a218c13f717c98be0fbada9/zirgen/circuit/bigint/elliptic_curve.cpp#L382 makes the curve parameters generic (always affine + Weierstrass though), allowing us to run this on other curves as long as we generate them.